src/source.cpp

```diff
@@ -180,19 +283,20 @@ SourceSite IndependentSource::sample(uint64_t* seed) const
   static int n_accept = 0;
 
   while (!found) {
-    // Set particle type
-    Particle p;
-    p.type() = particle_;
-    p.u() = {0.0, 0.0, 1.0};
 
     // Sample spatial distribution
-    p.r() = space_->sample(seed);
+    site.r = space_->sample(seed);
 
-    // Now search to see if location exists in geometry
-    found = exhaustive_find_cell(p);
+    // Check if sampled position satisfies spatial constraints
+    found = satisfies_spatial_constraints(site.r);
 
     // Check if spatial site is in fissionable material
```

Comment:
Let's move this fissionable check to the outer `Source` code. I really deeply believe that's the right thing to do here.

Why should an `IndependentSource` enjoy fissionable rejection but not a `FileSource`?